src/main/groovy/io/seqera/wave/service/builder/KubeBuildStrategy.groovy

```diff
@@ -130,12 +132,38 @@ class KubeBuildStrategy extends BuildStrategy {
     @Override
     void cleanup(BuildRequest req) {
         super.cleanup(req)
-        final name = podName(req)
+        final name = getName(req)
         try {
-            k8sService.deletePod(name)
+            if( req.formatDocker() )
+                k8sService.deleteJob(name)
+            else
+                k8sService.deletePod(name)
         }
         catch (Exception e) {
-            log.warn ("Unable to delete pod=$name - cause: ${e.message ?: e}", e)
+            log.warn ("Unable to delete job or pod with name=$name - cause: ${e.message ?: e}", e)
+        }
+    }
+
+    protected V1Pod  launchContainerBuild(String name, String buildImage, List<String> buildCmd, BuildRequest req, Path configFile, SpackConfig spackConfig, Map<String,String> nodeSelector){
+        if(req.formatSingularity()) {
+            return k8sService.buildContainer(name, buildImage, buildCmd, req.workDir, configFile, spackConfig, nodeSelector)
+        }else {
+            final job = k8sService.buildJob(name, buildImage, buildCmd, req.workDir, configFile, spackConfig, nodeSelector)
+            final podList = k8sService.waitJob(job, buildConfig. buildTimeout.toMillis())
+            final int size = podList?.items?.size()
+
+            if( size < 1 )
+                throw new BuildTimeoutException("Build job timed out with build id: $req.buildId")
+
+            // Find the latest created pod among the pods associated with the job
+            def latestPod = podList.getItems().get(0)
+            for (def pod : podList.items) {
+                if (pod.metadata.creationTimestamp.isAfter(latestPod.metadata.creationTimestamp)) {
+                    latestPod = pod
+                }
+            }
```

Comment:
Why Singularity uses a pod and docker a job?

Reply:
Yes, it would be better to keep them aligned. Along with this 

1. is the use of job really needed? could not be used just a pod ad it was done before .
2. if a job is needed, not getting what's needed to find out the underlying pod and wait for it. It should be possible to wait directly on the job status

Reply:
There is a rootles pod also https://github.com/moby/buildkit/blob/master/examples/kubernetes/pod.rootless.yaml
But earlier, you suggested using https://github.com/seqeralabs/wave/pull/503#issuecomment-2115650944

Underlying pods found using label `job=JOB_NAME`
https://github.com/seqeralabs/wave/blob/72b59f2a8a62326100324621384080142b18ee10/src/main/groovy/io/seqera/wave/service/k8s/K8sServiceImpl.groovy#L701

Yes, I think we can wait on the job status also

Reply:
I believe I've linked the job implementation only because it was the first thing I've found. I would go with pod in the final implementation both for singularity and docker format, provided it does not impact security requirements

Reply:
Thanks for the clarification. I will make the changes. 
implementation of pod and job is same, so there should not be any impact on security